packages/jest/src/automatic.ts

```diff
@@ -84,16 +86,113 @@ export async function automaticCheck(opts: AutoCheckOpts = defaultAutoCheckOpts)
     }
 }
 
+// https://thoughtspile.github.io/2018/06/20/serialize-promises/
+// function serializePromises(immediate: CallableFunction) {
+//     // This works as our promise queue
+//     let last = Promise.resolve();
+//     return function (...a) {
+//         // Catch is necessary here — otherwise a rejection in a promise will
+//         // break the serializer forever
+//         last = last.catch(() => {}).then(() => immediate(...a));
+//         return last;
+//     };
+// }
+
+// https://advancedweb.hu/how-to-serialize-calls-to-an-async-function/
+// const serializePromises = (fn: CallableFunction) => {
+//     // Start of the queue. Every other call is appended to this Promise.
+//     let queue = Promise.resolve();
+//     return (...args: any[]) => {
+//         // Adds the function call to the queue and saves its result.
+//         // Will be resolved when the current and all the previous calls are resolved.
+//         const result = queue.then(() => fn(...args));
+//         // Makes sure that the queue won’t get stuck in rejection when one part of it is rejected.
+//         queue = result.catch((err) => {
+//             console.log(err);
+//         });
+//         return result;
+//     };
+// };
+
+const mutexTimeout = 5000;
+const mutex = withTimeout(new Mutex(), mutexTimeout, new Error('Timed-out waiting for axe'));
+
+function observerCallback(mutations: MutationRecord[], _observer: MutationObserver) {
+    const violations: AxeResults = []; // TODO (refactor): move to global/test scope
+    for (const mutation of mutations) {
+        // log('Mutation event triggered on', mutation.target.nodeName);
+        for (const node of mutation.addedNodes) {
```

Comment:
@trevor-bliss I am running into the following error with trying to execute axe with a mutation observer 

```
Axe is already running.
 Use `await axe.run()` to wait for the previous run to finish before starting a new run.
          at getViolations (sa11y/packages/common/src/axe.ts:44:15)
          at getViolationsJSDOM (sa11y/packages/assert/src/assert.ts:32:12)
```

I have tried to
1. Serialize axe invocations using a Promise queue
2. Use array `.reduce()` to serialize axe invocation.
3. Wrap axe invocation with a mutex

None of them seem to fix the issue. Is there anything obvious that I am doing wrong? 
Can you please take a look at the code. Different variations of unsuccessful solutions are commented out.

Reply:
I don't see anything obviously wrong. Taking a step back, are we sure we want to run the checks at this fine grained of a level? 

I question how many a11y issues we would find in a unit test using MutationObserver compared to just running the check at the end of the test like we do now.

Reply:
Thanks Trevor. 

- It is to be seen how much more a11y coverage we will get with this approach. We can pare down the level to something that works both in terms of perf and num of a11y issues found. 
- Plan is to look at potential ways to detect 'significant' DOM changes by some set of predefined params once the basic workflow is working. 
- Also tests are ideally supposed to clean the DOM at the end, which a good chunk of tests do follow. So this approach will also increase a11y coverage for those tests.